### ISSUE TITLE
label-content-name-mismatch incorrectly categorizes emojis using Unicode text variation selector

### ISSUE THREAD

**Issue by lberrymage** [dequelabs/axe-core#5400]
### Product

axe-core

### Product Version

4.13.0

### Latest Version

- [x] I have tested the issue with the latest version of the product

#### Expectation

Evaluation of axe-core's label-content-name-mismatch rule on a link containing U+21A9 (LEFTWARDS ARROW WITH HOOK) followed by U+FE0E (VARIATION SELECTOR-15, [used for text presentation of emoji characters](https://www.unicode.org/reports/tr51/tr51-31.html#def_text_presentation_selector)) results in an outcome of "incomplete", just as it does for U+21A9 by itself as well as U+21A9 followed by U+FE0F (VARIATION SELECTOR-16, [used for emoji presentation of emoji characters](https://www.unicode.org/reports/tr51/tr51-31.html#def_emoji_presentation_selector)).

#### Actual

Evaluation of axe-core's label-content-name-mismatch rule in the described situation results in an outcome of "violations" even though it results in "incomplete" for U+21A9 by itself and U+21A9 followed by U+FE0F.

#### How to Reproduce

Copy the following code to an HTML file and open it in a web browser:

```html
<!doctype html>
<html lang="en">
<title>axe-core U+FE0E repro</title>
<script src="https://cdn.jsdelivr.net/npm/axe-core@4.13.0/axe.min.js"></script>
<!-- U+21A9 U+FE0E (text presentation): violation -->
<a id="text-vs" href="#" aria-label="Back to reference 1">&#x21A9;&#xFE0E;</a>
<!-- U+21A9 alone: no violation -->
<a id="no-vs" href="#" aria-label="Back to reference 1">&#x21A9;</a>
<!-- U+21A9 U+FE0F (emoji presentation): no violation -->
<a id="emoji-vs" href="#" aria-label="Back to reference 1">&#x21A9;&#xFE0F;</a>
<pre id="out"></pre>
<script>
    axe.run({ runOnly: "label-content-name-mismatch" }).then((r) => {
        out.textContent = ["violations", "incomplete", "passes"]
            .flatMap((k) => r[k].flatMap((rule) => rule.nodes.map((n) => `${n.target}: ${k}`)))
            .join("\n");
    });
</script>
```

You should see the following output in the browser windowt:

[↩︎](#) [↩](#) [↩️](#)
#text-vs: violations
#no-vs: incomplete
#emoji-vs: incomplete

#### Additional context

I found this issue while running Lighthouse audits on a website which uses U+21A9 + U+FE0E for footnote backlinks to explicitly choose the text variant of the emoji. The bug appears to be in `removeUnicode()`, as `emojiRegexText()` doesn't select U+FE0E after an emoji character. The text variant may not technically be an emoji, but it is a symbol, so `removeUnicode()` should probably remove it.

In general, I think it probably makes sense for `removeUnicode()` to remove any variation selectors that apply to characters otherwise being removed, as leaving only the variation selector behind isn't meaningful.

**Comment by maricastroc** [comment]
I reproduced this on the current develop branch and traced it to variation selectors not being removed by removeUnicode(). In the FE0E case, the base character is removed by the emoji regex but FE0E remains, which causes isHumanInterpretable to treat the content as meaningful and also affects tokenization.

I prototyped a small fix by adding a variation-selector regex covering the Unicode Variation_Selector ranges and applying it in the nonBmp branch of removeUnicode(). This fixes the reported FE0E case as well as FE00, supplementary ideographic variation selectors, and selectors following ordinary letters/digits. The existing unit, integration, virtual-rules and ACT 2ee8b8 tests continue to pass.

Before opening a PR, I'd like to confirm the intended scope: would you prefer the fix to be limited to removeUnicode(), or should curateTokens also strip variation selectors to avoid splitting ideographic variation sequences? And should hasUnicode() be updated for symmetry as was done for Cf characters in #4102?

Happy to open a PR with the regression coverage if that direction sounds right.
